src/panels/SkyBrowserPanel/WorldWideTelescope/hooks.ts

```diff
@@ -0,0 +1,103 @@
+import { useEffect } from 'react';
+
+import { useAppSelector } from '@/redux/hooks';
+import { useWindowSize } from '@/windowmanagement/Window/hooks';
+
+import {
+  useSelectedBrowserColor,
+  useSelectedBrowserCoords,
+  useSelectedBrowserFov,
+  useSelectedBrowserRadius,
+  useSkyBrowserSelectedImages,
+  useSkyBrowserSelectedOpacities
+} from '../hooks';
+
+import { useWwtProvider } from './WwtProvider/hooks';
+
+// These are the hooks that will keep tabs of the redux state and
+// when it changes, it will pass along these messages to WWT
+
+// The hooks are: images, opacities, aim, border color and border radius
+export function useUpdateSelectedImages() {
+  const { imageCollectionLoaded, loadImage, removeImage } = useWwtProvider();
+  const imageList = useAppSelector((state) => state.skybrowser.imageList);
+  const selectedImages = useSkyBrowserSelectedImages();
+
+  // Update images in WWT when the selected images changes
+  useEffect(() => {
+    if (
+      imageList.length === 0 ||
+      !imageCollectionLoaded ||
+      selectedImages === undefined
+    ) {
+      return;
+    }
+    // Brute force this as the performance loss is negligible and there are many complicated cases
+    selectedImages.toReversed().map((index) => {
+      loadImage(imageList[index]?.url);
+    });
+    return () => selectedImages?.forEach((image) => removeImage(imageList[image].url));
+  }, [imageList, selectedImages, imageCollectionLoaded, loadImage, removeImage]);
+}
+
+export function useUpdateOpacities() {
+  const { imageCollectionLoaded, setOpacity } = useWwtProvider();
+  const imageList = useAppSelector((state) => state.skybrowser.imageList);
+  const selectedImages = useSkyBrowserSelectedImages();
+  const opacities = useSkyBrowserSelectedOpacities();
+
+  // Update opacities in WWT when the opacities changes
+  useEffect(() => {
+    if (imageList.length === 0 || !imageCollectionLoaded || opacities === undefined) {
+      return;
+    }
+    // Brute force this as the performance loss is negligible and there are many complicated cases
+    opacities.map((opacity, i) => {
+      if (!selectedImages) {
```

Comment:
Seems like this check should be moved to outside the .map call.  Part of the if statement above, maybe?

Also, seems like we expect `selectedImages` and `opacities` to have the same size.  Should these two things really be one combined data structure?  To avoid the direct dependency between the two lists.... I assume that `opacities` is the opacity setting for each of the selected images?